README.md

```diff
@@ -14,3 +14,94 @@ npm start
 This project uses npm scripts to build the site to the `dist` directory.
 
 - `npm start` - Build site, watch for changes and run a hot code reloading server on http://localhost:3000
+
+## Helpers
+
+Any module dropped into the *helpers* folder will be passed in the `content` variable to all templates, with the obvious camel-case conversion (i.e. `is-equal` would be available as `content.isEqual`).
+
+## Declaring schemas
+
+Fields in a *content.json* or *facts.json* which do not have a declared `type` in the corresponding *schema.json* are assumed to be of type *text*.  The available types are:
+
+* collection
+* color
+* img
+* list
+* map
+* text
+* textarea
+
+If validation beyond that which is built in for these fields is required, it can be supplied in a `validation` key in the *schema.json*:
+
+```json
+{
+  "exampleField": {
+    "type": "text",
+    "validation": {
+      "pattern": "^(https?://)?([0-9a-z.-]+).([a-z.]{2,6})([/[0-9][a-z].-]*)*/?$"  
+    }
+  },  
+}
+```
+
+Available validation options are as per JSON-schema and can be found [here](http://json-schema.org/latest/json-schema-validation.html).
+
+### Regex validation
+
+JSON-schema regex patterns only allow a subset of those available in Javascript - for example, periods will match the period character rather than any character.  Full details can be found under the *pattern* section [here](http://json-schema.org/latest/json-schema-validation.html).
+
+Some useful JSON-schema regex patterns:
+
+**URL** - `^(https?://)?([0-9a-z.-]+).([a-z.]{2,6})([/[0-9][a-z].-]*)*/?$`
+
+## Defining collections
+
+A collection (array of objects) is declared in the appropriate *schema.json* as a **one-element array** containing the declaration object rather than the object itself.  Within this single object, declarations are as they would be for any other field.  For example:
+
+```json
+{
+  "carousel": [{
+    "title": {
+      "type": "text",
+      "default": "New entry"
+     },
+    "image": {
+      "type": "img",
+      "default": "http://placehold.it/350x150"
+    },
+    "link": {
+      "type": "text",
+      "default": "http://example.com",
+      "validation": {
+        "pattern": "^(https?://)?([0-9a-z.-]+).([a-z.]{2,6})([/[0-9][a-z].-]*)*/?$"  
+      }
+    }
+  }]
+}
+```
+
+**Note that in a collection every field requires a `default` to be set for the client to be able to add new items.**
```

Comment:
As discussed, it would be nice if the default was optional. Obviously not something to fix in this repo or PR...